libraries/AP_BattMonitor/AP_BattMonitor.h

```diff
@@ -18,16 +18,18 @@
 class AP_BattMonitor_Backend;
 class AP_BattMonitor_Analog;
 class AP_BattMonitor_SMBus;
-class AP_BattMonitor_SMBus_I2C;
+class AP_BattMonitor_SMBus_Solo;
 class AP_BattMonitor_SMBus_PX4;
+class AP_BattMonitor_SMBus_Maxcell;
 
 class AP_BattMonitor
 {
     friend class AP_BattMonitor_Backend;
     friend class AP_BattMonitor_Analog;
     friend class AP_BattMonitor_SMBus;
-    friend class AP_BattMonitor_SMBus_I2C;
+    friend class AP_BattMonitor_SMBus_Solo;
     friend class AP_BattMonitor_SMBus_PX4;
+    friend class AP_BattMonitor_SMBus_Maxcell;
```

Comment:
I think that it is unnecessary if it can distinguish by the value of I2C registers.